test/distributed/test_inductor_collectives.py

```diff
@@ -48,6 +48,38 @@ def world_size(self) -> int:
         # works around issue with skipif<2 and workers with unpredictable #s gpu
         return 2
 
+    @unittest.skipIf(not has_triton(), "Inductor+gpu needs triton and recent GPU arch")
+    @skip_if_lt_x_gpu(2)
+    # TODO: somehow inductor bg compile threads are causing hangs at exit with distributed work dtor
+    @patch.object(torch._inductor.config, "compile_threads", 1)
+    def test_broadcast_inductor(self):
+        """
+        Testing if broadcast works correctly when using inductor
+        """
+
+        def example(tensor, src, *, tag, ranks, group_size):
+            res = torch.ops.c10d_functional.broadcast(tensor, src, tag, ranks, group_size)
+            res = torch.ops.c10d_functional.wait_tensor(res)
+            return res
+
+        def compile(func, example_inputs):
+            graph = make_fx(func)(*example_inputs)
+            return inductor_compile_fx(graph, example_inputs)
+
+        with _dynamo_dist_per_rank_init(self.rank, self.world_size):
+
+            example = functools.partial(
+                example,
+                **self.get_world_trs(),
+            )
+            inputs = (
+                    torch.ones(4, 4, device="cuda") if self.rank == 0 else torch.zeros(4,4, device="cuda"),
+                    0
+                )
+            compiled_func = compile(example, inputs)
+            compiled_out = compiled_func(*inputs)
+            self.assertTrue(same(torch.ones(4,4, device="cuda"), compiled_out))
```

Comment:
nit: usually its good to actually run the 'example' in eager mode and then compare its output rather than constructing the expected output.

and it might be worth using more interesting sets of values than zeros/ones since in a lot of cases feeding zeros through a broken program can fail to catch bugs. in this case im not too worried about it since we know the underlying collective is well tested..